source/tokens.rs

```diff
@@ -70,6 +78,70 @@ token!(
     "The `AS` token.\n\nRepresent the alias operator, e.g. `use Foo\\Bar as Baz`."
 );
 token!(
+    ASSIGN: "=";
+    "The `ASSIGN` token.\n\nRepresent a binding of a value to a variable, e.g. `$x = 42`."
+);
+token!(
+    ASSIGN_REFERENCE: "=&";
```

Comment:
Are you sure this should be a token ? It will be difficult to support notation like `= & $bla` with a single token.